changesummary.py

```diff
@@ -0,0 +1,268 @@
+        """
+        commit_type = 'feat' if is_feature else 'fix'
+        breaking_change_char = '!' if is_breaking else ''
+        return '{0}({1}){2}: update the api\n'.format(commit_type, api_name, \
+                                                        breaking_change_char)
+
+    def _get_keys_to_ignore(self):
+        """ Returns keys to ignore as an array of strings.
+            args: None
+        """
+        keys_to_ignore = [
+            'description',
+            'documentation',
+            'enum',
+            'etag',
+            'revision',
+            'title'
+            'url',
+            'rootUrl'
+            ]
+        return keys_to_ignore
+
+    def _get_summary_and_write_to_disk(self, dataframe, directory):
+        """Writes summary information to file about changes made to discovery
+            artifacts based on the provided dataframe and returns a dataframe
+            with the same. The file `'allapis.summary'` is saved to the current
+            working directory.
+            args:
+                dataframe: a pandas dataframe containing summary change
+                    information for all discovery artifacts
+                directory: path where the summary file should be saved
+        """
+        dataframe['IsFeature'] = np.where( \
+            (dataframe['ChangeType'] == ChangeType.DELETED) | \
+                (dataframe['ChangeType'] == ChangeType.ADDED), True, np.nan)
+
+        dataframe['IsBreaking'] = np.where( \
+            (dataframe['ChangeType'] == ChangeType.DELETED), True, np.nan)
+
+        dataframe['IsFeatureAggregate'] = dataframe.groupby( \
+            'Name').IsFeature.transform(lambda x : x.any())
+
+        dataframe['IsBreakingAggregate'] = dataframe.groupby(\
+            'Name').IsBreaking.transform(lambda x : x.any())
+
+        dataframe['Summary'] = np.vectorize(self._build_summary_message)\
+                                            (dataframe['Name'],
+                                            dataframe['IsFeatureAggregate'],
+                                            dataframe['IsBreakingAggregate'])
+
+        with open("".join([directory,"/", "allapis.summary"]), 'w') as f:
+            f.writelines([summary_msg for summary_msg in dataframe.Summary.unique()])
+
+        return dataframe
+
+    def _write_verbose_changes_to_disk(self, dataframe, directory, summary_df):
+        """"Writes verbose information to file about changes made to discovery
+            artifacts based on the provided dataframe. A separate file is saved
+            for each api in the current working directory. The extension of the
+            files will be `'.verbose'`.
+
+            args:
+                dataframe: a pandas dataframe containing verbose change
+                    information for all discovery artifacts
+                directory: path where the summary file should be saved
+                summary_df: A dataframe containing a summary of the changes
```

Comment:
Please add types to these args. E.g., `directory (str): path where the summary file should be saved`

Reply:
Done.